cmake/CMakeLists.txt

```diff
@@ -162,12 +162,23 @@ endif()
 
 if(ENABLE_MSCG OR ENABLE_USER-ATC OR ENABLE_USER-AWPMD OR ENABLE_USER-QUIP)
   find_package(LAPACK)
-  if(LAPACK_FOUND)
-    list(APPEND LAMMPS_LINK_LIBS ${LAPACK_LIBRARIES})
-  else()
+  if(NOT LAPACK_FOUND)
     enable_language(Fortran)
     file(GLOB LAPACK_SOURCES ${LAMMPS_LIB_SOURCE_DIR}/linalg/*.f)
-    list(APPEND LIB_SOURCES ${LAPACK_SOURCES})
+
+    if(BUILD_SHARED_PKG_LIBS)
+      add_library(linalg SHARED ${LAPACK_SOURCES})
+      set_target_properties(${PKG_LIB} PROPERTIES SOVERSION ${SOVERSION})
+      if(INSTALL_LIB)
+        install(TARGETS linalg LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+      else()
+        message(FATAL_ERROR "Shared library has to be installed, use -DINSTALL_LIB=ON to install lammps with a library")
+      endif()
+    else()
+      add_library(linalg STATIC ${LAPACK_SOURCES})
+    endif()
+    target_compile_options(linalg PRIVATE -fno-second-underscore)
```

Comment:
Why do we need that?

Reply:
It serves as an example of how to add package specific compile options and I added it to make it more like the lib Makefile. And it's just one of many such changes I see coming.

Reply:
Ok, examples are good, but why do want to append a gcc-only option by default? Is there a need to not have the second underscore?

Reply:
It looks like it only mattered with older compilers (gcc 3.x). Anyway, you are right that it has to be more selective wherever we put something like this. That was also the motivation why to use `target_include_directories` vs. manipulating compiler flags directly.

So in the end, we'll have to have an `if-else` to select options depending on the compiler. Each library might have different defaults.

Reply:
Sorry, but that sounds horribile, I really not trying to re-implement every details and default of the old build system into CMake!

Can you elaborate a bit more what the problem with the 2nd underscore is? (`libatc` and other `linalg` consumers have hard-codes function calls, right?) CMake can detect the Fortran name mangling automatically and then we could flag all compilers which append a second underscore as unsupported.

Reply:
Providing reasonable defaults to our users is part of the mandate. Sorry, we just have to be smart about it to not make it too messy (e.g. use variables set from outside via a script).

AFAIK it's about the older compilers adding a second underscore by default which then leads to a missing symbol during linking because the C/C++ code is looking for something with one underscore.

Reply:
I understand, but adding a flag, which isn't support but some modern Fortran compilers (e.g. flang) without an option to not do so, just not very smart.

The best would be to use CMake's `FortranCInterface()` interface, but that would involve touching actual code. So a good intermediate solution would be to check if the compiler supports `-fno-second-underscope` and only add it then. Plus, we will need to do the same for all Fortran packages, i.e. reax, meam, USER-QUIP and USER-QMMM, I guess.